velox/dwio/common/SelectiveColumnReader.h

```diff
@@ -509,6 +509,18 @@ class SelectiveColumnReader {
     return false;
   }
 
+  template <typename Decoder, typename ColumnVisitor>
+  void decodeWithVisitor(void* intDecoder, ColumnVisitor& visitor) {
+    const uint64_t* nulls =
+        nullsInReadRange_ ? nullsInReadRange_->as<uint64_t>() : nullptr;
+    auto decoder = reinterpret_cast<Decoder*>(intDecoder);
```

Comment:
Can we do a `dynamic_cast` then check the result is not null

Reply:
done